Library/Homebrew/dependency.rb

```diff
@@ -145,3 +145,31 @@ def installed?
     false
   end
 end
+
+class PrioritizedTapDependency < Dependency
+  attr_reader :tap
+
+  def initialize(name, tags=[], env_proc=DEFAULT_ENV_PROC, option_name=name)
+    @tap, _, option_name = option_name.rpartition "/"
+    begin
+      formula = Formulary.factory(name)
+    rescue FormulaUnavailableError
+      formula = Formulary.factory(option_name)
+      unless formula.core_formula?
+        opoo <<-EOS.undent
+          #{option_name} is required but is not provided by either #{@tap} or core.
+          We are installing from #{formula.tap} instead.
+          You may wish to contact the author of #{@tap} to report this.
+        EOS
+      end
+    end
```

Comment:
Somehow it does not work if I move it... I think I have some misunderstanding about dependency resolve procedure and I am looking for the problem. Sorry for that :(